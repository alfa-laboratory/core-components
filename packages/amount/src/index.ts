import { Amount as DefaultAmount } from './component';
import { PureAmount } from './pure/component';

type AmountType = typeof DefaultAmount & { Pure: typeof PureAmount };
export const Amount: AmountType = DefaultAmount as AmountType;
Amount.Pure = PureAmount;

export * from './types';
