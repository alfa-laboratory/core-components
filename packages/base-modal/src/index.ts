export * from './Component';
export * from './utils';
