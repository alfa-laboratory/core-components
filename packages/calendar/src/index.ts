export * from './Component';
export { usePeriod } from './usePeriod';
export { useCalendar } from './useCalendar';
export {
    russianWeekDay,
    dateInLimits,
    limitDate,
    monthName,
    DAYS_IN_WEEK,
    MONTHS,
    MONTHS_IN_YEAR,
    SUNDAY_INDEX,
    WEEKDAYS,
} from './utils';
