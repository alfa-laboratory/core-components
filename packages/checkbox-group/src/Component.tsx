import React, {
    FC,
    ReactNode,
    Children,
    cloneElement,
    ReactElement,
    ChangeEvent,
    MouseEvent,
    isValidElement,
} from 'react';
import cn from 'classnames';

import styles from './index.module.css';

export type Direction = 'horizontal' | 'vertical';
export type CheckboxGroupType = 'checkbox' | 'tag';

export type CheckboxGroupProps = {
    /**
     * Заголовок группы
     */
    label?: ReactNode;

    /**
     * Направление
     */
    direction?: Direction;

    /**
     * Тип компонента
     */
    type?: CheckboxGroupType;

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Текст ошибки
     */
    error?: string;

    /**
     * Дочерние элементы. Ожидаются компоненты `Checkbox` или `Tag`
     */
    children: ReactNode;

    /**
     * Обработчик изменения значения 'checked' одного из дочерних компонентов
     */
    onChange?: (
        event?: ChangeEvent | MouseEvent,
        payload?: {
            checked: boolean;
            name?: string;
        },
    ) => void;

    /**
     * Управление возможностью изменения состояния 'checked' дочерних компонентов CheckBox
     */
    disabled?: boolean;

    /**
     * Идентификатор для систем автоматизированного тестирования
     */
    dataTestId?: string;
};

export const CheckboxGroup: FC<CheckboxGroupProps> = ({
    children,
    className,
    direction = 'vertical',
    label,
    error,
    onChange,
    type = 'checkbox',
    dataTestId,
    disabled = false,
}) => {
    const renderCheckbox = (child: ReactElement) => {
        const { name, checked, className: childClassName } = child.props;

        const handleChange = (event: ChangeEvent) => {
            if (onChange) {
                onChange(event, { name, checked: !checked });
            }
        };

        return cloneElement(child, {
            onChange: handleChange,
            disabled,
            ...child.props,
            className: cn(childClassName, styles.checkbox),
        });
    };

    const renderTag = (child: ReactElement) => {
        const { name, checked } = child.props;

        const handleChange = (event: ChangeEvent | MouseEvent) => {
            if (onChange) {
                onChange(event, { name, checked: !checked });
            }
        };

        const clone = cloneElement(child, { onClick: handleChange, disabled, ...child.props });

        return (
            // eslint-disable-next-line jsx-a11y/label-has-associated-control
            <label className={cn(styles.checkbox, styles.tagLabel)}>
                {clone}
                <input
                    type='checkbox'
                    autoComplete='off'
                    onChange={handleChange}
                    disabled={disabled || child.props.disabled}
                    checked={checked}
                    className={styles.hiddenInput}
                />
            </label>
        );
    };

    return (
        <div
            className={cn(
                styles.component,
                styles[direction],
                { [styles.error]: error },
                className,
            )}
            data-test-id={dataTestId}
        >
            {label ? <span className={styles.label}>{label}</span> : null}

            {children ? (
                <div className={cn(styles.checkboxList)}>
                    {Children.map(children, child => {
                        if (isValidElement(child)) {
                            return type === 'checkbox' ? renderCheckbox(child) : renderTag(child);
                        }

                        return null;
                    })}
                </div>
            ) : null}

            {error && <span className={styles.errorMessage}>{error}</span>}
        </div>
    );
};
