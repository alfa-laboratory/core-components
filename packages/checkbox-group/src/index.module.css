@import '../../themes/src/default.css';

.component {
    display: flex;
    flex-direction: column;
}

.error {
    padding-left: var(--gap-m);
    border-left: 1px solid var(--color-red-brand-85-flat);
}

.checkbox-list {
    display: flex;
}

.vertical .checkbox-list {
    flex-direction: column;
    align-items: flex-start;
}

.vertical .checkbox {
    margin-bottom: var(--gap-m);

    &:last-child {
        margin-bottom: 0;
    }
}

.horizontal .checkbox-list {
    flex-wrap: wrap;
}

.horizontal .checkbox {
    margin-right: var(--gap-xl);
    margin-bottom: var(--gap-xs);

    &:last-child {
        margin-right: 0;
    }
}

.label {
    @mixin system_14-18_regular;
    margin-bottom: var(--gap-m);
    color: var(--color-dark-indigo-60-flat);
}

.error-message {
    @mixin system_14-18_regular;
    margin-top: var(--gap-m);
    color: var(--color-red-brand-85-flat);
}

.hidden-input {
    position: absolute;
    z-index: -1;
    top: 0;
    left: 0;
    opacity: 0;
}

.tag-label {
    position: relative;
}
