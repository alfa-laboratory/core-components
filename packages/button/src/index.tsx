import React, { forwardRef } from 'react';
import { Button as ButtonBase, ButtonProps } from './Component';

import styles from './index.module.css';

const Button = forwardRef<HTMLAnchorElement | HTMLButtonElement, ButtonProps>((props, ref) => {
    return <ButtonBase {...props} ref={ref} styles={styles} />;
});

export { Button, ButtonProps };
