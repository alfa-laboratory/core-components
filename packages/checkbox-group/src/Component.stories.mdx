import { Meta, Story, Props, Preview, Title } from '@storybook/addon-docs/blocks';
import { text, boolean, select } from '@storybook/addon-knobs';
import { Status } from 'storybook/blocks/status';
import { Design } from 'storybook/blocks/design';

import { CheckboxGroup } from './Component';
import { name, version } from '../package.json';
import { Checkbox } from '../../checkbox/src';
import { Tag } from '../../tag/src';


<Meta title='Компоненты|CheckboxGroup' component={CheckboxGroup} />

<Title>
    CheckboxGroup ({name}@{version})
</Title>

<Status>
  Candidate Recommendation
</Status>

```tsx
import { CheckboxGroup } from '@alfalab/core-components-checkbox-group';
```

<Design>
  <a href="https://www.figma.com/file/hqSP3L6qu8UcL3sf18Su1M/Web-Components?node-id=83%3A306">Figma</a>
</Design>

## Описание

Компонент группы чекбоксов.

<Story name='Песочница'>
    {React.createElement(() => {
        const [value, setValue] = React.useState({ one: false, two: false, three: false });
        const onChange = (_, payload) => {
            setValue({ ...value, [payload.name]: payload.checked });
        };
        const direction = select('direction', ['vertical', 'horizontal'], 'vertical');
        const error = text('error', '');
        const disabled = boolean('disabled', false);
        const label = text('label', 'Заголовок группы');
        return (
            <CheckboxGroup
                label={label}
                onChange={onChange}
                direction={direction}
                error={error}
                disabled={disabled}
            >
                <Checkbox label='Первый вариант' name='one' checked={value.one} />
                <Checkbox label='Второй вариант' name='two' checked={value.two} />
                <Checkbox label='Третий вариант' name='three' checked={value.three} />
            </CheckboxGroup>
        );
    })}
</Story>

<Props of={CheckboxGroup} />

### Группа чекбоксов, использующая компонент `Tag`

<Preview>
    {React.createElement(() => {
        const [value, setValue] = React.useState({ one: false, two: false, three: false });
        const onChange = (_, payload) => {
            setValue({ ...value, [payload.name]: payload.checked });
        };
        return (
            <CheckboxGroup
                label="Заголовок группы"
                onChange={onChange}
                direction="horizontal"
                type='tag'
            >
                <Tag name='one' checked={value.one}>
                    Первый вариант
                </Tag>
                <Tag name='two' checked={value.two}>
                    Второй вариант
                </Tag>
                <Tag name='three' checked={value.three}>
                    Третий вариант
                </Tag>
            </CheckboxGroup>
        );
    })}
</Preview>
