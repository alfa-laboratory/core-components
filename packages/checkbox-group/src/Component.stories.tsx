import React, { useState } from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import { withDesign } from 'storybook-addon-designs';

import { CheckboxGroup } from './Component';
import { Checkbox } from '../../checkbox/src';
import { Tag } from '../../tag/src';

export default {
    title: 'Common|CheckboxGroup',
    component: CheckboxGroup,
    decorators: [withDesign, withKnobs],
};

export const CheckboxStory = () => {
    const [value, setValue] = useState({ one: false, two: false, three: false });

    const onChange = (_, payload) => {
        setValue({ ...value, [payload.name]: payload.checked });
    };

    return (
        <CheckboxGroup label='Заголовок группы' onChange={onChange}>
            <Checkbox label='Первый вариант' name='one' checked={value.one} />

            <Checkbox label='Второй вариант' name='two' checked={value.two} />

            <Checkbox label='Третий вариант' name='three' checked={value.three} />
        </CheckboxGroup>
    );
};

export const CheckboxTagStory = () => {
    const [value, setValue] = useState({ one: false, two: false, three: false });

    const onChange = (_, payload) => {
        setValue({ ...value, [payload.name]: payload.checked });
    };

    return (
        <CheckboxGroup
            label='Заголовок группы'
            onChange={onChange}
            direction='horizontal'
            type='tag'
        >
            <Tag name='one' checked={value.one}>
                Первый вариант
            </Tag>

            <Tag name='one' checked={value.two}>
                Второй вариант
            </Tag>

            <Tag name='one' checked={value.three}>
                Третий вариант
            </Tag>
        </CheckboxGroup>
    );
};

CheckboxStory.story = {
    name: 'CheckboxGroup',
    parameters: {},
};

CheckboxTagStory.story = {
    name: 'CheckboxTagGroup',
    parameters: {},
};
