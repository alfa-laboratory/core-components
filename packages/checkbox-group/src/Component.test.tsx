import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { CheckboxGroup, CheckboxGroupProps } from './index';
import { Checkbox } from '../../checkbox/src';
import { Tag } from '../../tag/src';

const Group = ({ ...restProps }: Partial<CheckboxGroupProps>) => (
    <CheckboxGroup label='Заголовок группы' {...restProps}>
        <Checkbox label='Первый вариант' name='one' className='my-checkbox' />

        <Checkbox label='Второй вариант' name='two' />

        <Checkbox label='Третий вариант' name='three' />
    </CheckboxGroup>
);

const TagGroup = ({ ...restProps }: Partial<CheckboxGroupProps>) => (
    <CheckboxGroup label='Заголовок группы' type='tag' {...restProps}>
        <Tag name='one' className='my-tag'>
            Первый вариант
        </Tag>

        <Tag name='two'>Второй вариант</Tag>

        <Tag name='three'>Третий вариант</Tag>
    </CheckboxGroup>
);

describe('Checkbox', () => {
    describe('Display tests', () => {
        it('should display with children like boolean or string or others react children type correctly', () => {
            expect(
                render(
                    <CheckboxGroup label='Заголовок группы'>
                        {0 && (
                            <Checkbox label='Первый вариант' name='one' className='my-checkbox' />
                        )}
                        {false && (
                            <Checkbox label='Первый вариант' name='one' className='my-checkbox' />
                        )}
                        {null && (
                            <Checkbox label='Первый вариант' name='one' className='my-checkbox' />
                        )}
                        {'' && (
                            <Checkbox label='Первый вариант' name='one' className='my-checkbox' />
                        )}
                    </CheckboxGroup>,
                ),
            ).toMatchSnapshot();
        });

        it('should display checkbox group with one child correctly', () => {
            expect(
                render(
                    <CheckboxGroup label='Заголовок группы'>
                        <Checkbox label='Первый вариант' name='one' className='my-checkbox' />
                    </CheckboxGroup>,
                ),
            ).toMatchSnapshot();
        });

        it('should display checkbox group correctly', () => {
            expect(render(<Group />)).toMatchSnapshot();
        });

        it('should display tag group correctly', () => {
            expect(render(<TagGroup />)).toMatchSnapshot();
        });

        it('should display checkbox group with error correctly', () => {
            expect(render(<Group error='Error' />)).toMatchSnapshot();
        });
    });

    describe('Styles tests', () => {
        it('should set custom class', () => {
            const className = 'custom-class';
            const { container } = render(<Group className={className} />);

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should set `vertical` class', () => {
            const { container } = render(<Group direction='vertical' />);

            expect(container.firstElementChild).toHaveClass('vertical');
        });

        it('should set `error` class', () => {
            const { container } = render(<Group error='Error message' />);

            expect(container.firstElementChild).toHaveClass('error');
        });
    });

    describe('Attributes tests', () => {
        it('should set `data-test-id` atribute', () => {
            const dataTestId = 'test-id';
            const { container } = render(<Group dataTestId={dataTestId} />);

            const testIdAttr = container.firstElementChild?.getAttribute('data-test-id');

            expect(testIdAttr).toBe(dataTestId);
        });
    });

    describe('Render tests', () => {
        it('should unmount without errors', () => {
            const { unmount } = render(<Group />);

            expect(unmount).not.toThrowError();
        });

        it('should contain label', () => {
            const labelText = 'I am label';

            const { getByText } = render(<Group label={labelText} />);

            expect(getByText(labelText)).toBeInTheDocument();
        });

        it('should contain error message', () => {
            const errorText = 'I am error';

            const { getByText } = render(<Group error={errorText} />);

            expect(getByText(errorText)).toBeInTheDocument();
        });

        it('should disable all checkboxes if `disabled` prop is present', () => {
            const { container } = render(<Group disabled={true} />);

            const inputs = container.querySelectorAll('input');

            let allDisabled = true;

            inputs.forEach(input => {
                if (!input.disabled) {
                    allDisabled = false;
                }
            });

            expect(allDisabled).toBe(true);
        });
    });
});

describe('Interaction tests', () => {
    test('should call `onChange` callback if click on checkbox', () => {
        const cb = jest.fn();

        const { container } = render(<Group onChange={cb} />);

        const checkbox = container.querySelector('.my-checkbox');

        if (checkbox) {
            fireEvent.click(checkbox);
        }

        expect(cb).toBeCalledTimes(1);
    });

    test('should call `onChange` callback if click on tag', () => {
        const cb = jest.fn();

        const { container } = render(<TagGroup onChange={cb} />);

        const tag = container.querySelector('.my-tag');

        if (tag) {
            fireEvent.click(tag);
        }

        expect(cb).toBeCalledTimes(1);
    });
});
