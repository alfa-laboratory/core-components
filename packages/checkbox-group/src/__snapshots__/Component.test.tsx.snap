// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Display tests should display checkbox group correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component checkbox vertical"
      >
        <span
          class="label"
        >
          Заголовок группы
        </span>
        <div
          class="checkboxList"
        >
          <label
            class="component my-checkbox checkbox"
          >
            <span
              class="box"
            >
              <input
                type="checkbox"
              />
            </span>
            <span
              class="content"
            >
              <span
                class="label"
              >
                Первый вариант
              </span>
            </span>
          </label>
          <label
            class="component checkbox"
          >
            <span
              class="box"
            >
              <input
                type="checkbox"
              />
            </span>
            <span
              class="content"
            >
              <span
                class="label"
              >
                Второй вариант
              </span>
            </span>
          </label>
          <label
            class="component checkbox"
          >
            <span
              class="box"
            >
              <input
                type="checkbox"
              />
            </span>
            <span
              class="content"
            >
              <span
                class="label"
              >
                Третий вариант
              </span>
            </span>
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component checkbox vertical"
    >
      <span
        class="label"
      >
        Заголовок группы
      </span>
      <div
        class="checkboxList"
      >
        <label
          class="component my-checkbox checkbox"
        >
          <span
            class="box"
          >
            <input
              type="checkbox"
            />
          </span>
          <span
            class="content"
          >
            <span
              class="label"
            >
              Первый вариант
            </span>
          </span>
        </label>
        <label
          class="component checkbox"
        >
          <span
            class="box"
          >
            <input
              type="checkbox"
            />
          </span>
          <span
            class="content"
          >
            <span
              class="label"
            >
              Второй вариант
            </span>
          </span>
        </label>
        <label
          class="component checkbox"
        >
          <span
            class="box"
          >
            <input
              type="checkbox"
            />
          </span>
          <span
            class="content"
          >
            <span
              class="label"
            >
              Третий вариант
            </span>
          </span>
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox Display tests should display checkbox group with error correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component checkbox vertical error"
      >
        <span
          class="label"
        >
          Заголовок группы
        </span>
        <div
          class="checkboxList"
        >
          <label
            class="component my-checkbox checkbox"
          >
            <span
              class="box"
            >
              <input
                type="checkbox"
              />
            </span>
            <span
              class="content"
            >
              <span
                class="label"
              >
                Первый вариант
              </span>
            </span>
          </label>
          <label
            class="component checkbox"
          >
            <span
              class="box"
            >
              <input
                type="checkbox"
              />
            </span>
            <span
              class="content"
            >
              <span
                class="label"
              >
                Второй вариант
              </span>
            </span>
          </label>
          <label
            class="component checkbox"
          >
            <span
              class="box"
            >
              <input
                type="checkbox"
              />
            </span>
            <span
              class="content"
            >
              <span
                class="label"
              >
                Третий вариант
              </span>
            </span>
          </label>
        </div>
        <span
          class="errorMessage"
        >
          Error
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component checkbox vertical error"
    >
      <span
        class="label"
      >
        Заголовок группы
      </span>
      <div
        class="checkboxList"
      >
        <label
          class="component my-checkbox checkbox"
        >
          <span
            class="box"
          >
            <input
              type="checkbox"
            />
          </span>
          <span
            class="content"
          >
            <span
              class="label"
            >
              Первый вариант
            </span>
          </span>
        </label>
        <label
          class="component checkbox"
        >
          <span
            class="box"
          >
            <input
              type="checkbox"
            />
          </span>
          <span
            class="content"
          >
            <span
              class="label"
            >
              Второй вариант
            </span>
          </span>
        </label>
        <label
          class="component checkbox"
        >
          <span
            class="box"
          >
            <input
              type="checkbox"
            />
          </span>
          <span
            class="content"
          >
            <span
              class="label"
            >
              Третий вариант
            </span>
          </span>
        </label>
      </div>
      <span
        class="errorMessage"
      >
        Error
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox Display tests should display checkbox group with one child correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component checkbox vertical"
      >
        <span
          class="label"
        >
          Заголовок группы
        </span>
        <div
          class="checkboxList"
        >
          <label
            class="component my-checkbox checkbox"
          >
            <span
              class="box"
            >
              <input
                type="checkbox"
              />
            </span>
            <span
              class="content"
            >
              <span
                class="label"
              >
                Первый вариант
              </span>
            </span>
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component checkbox vertical"
    >
      <span
        class="label"
      >
        Заголовок группы
      </span>
      <div
        class="checkboxList"
      >
        <label
          class="component my-checkbox checkbox"
        >
          <span
            class="box"
          >
            <input
              type="checkbox"
            />
          </span>
          <span
            class="content"
          >
            <span
              class="label"
            >
              Первый вариант
            </span>
          </span>
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox Display tests should display tag group correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component tag vertical"
      >
        <span
          class="label"
        >
          Заголовок группы
        </span>
        <div
          class="checkboxList"
        >
          <label
            class="checkbox tagLabel"
          >
            <button
              class="component s my-tag"
              type="button"
            >
              <span>
                Первый вариант
              </span>
            </button>
            <input
              autocomplete="off"
              class="hiddenInput"
              type="checkbox"
            />
          </label>
          <label
            class="checkbox tagLabel"
          >
            <button
              class="component s"
              type="button"
            >
              <span>
                Второй вариант
              </span>
            </button>
            <input
              autocomplete="off"
              class="hiddenInput"
              type="checkbox"
            />
          </label>
          <label
            class="checkbox tagLabel"
          >
            <button
              class="component s"
              type="button"
            >
              <span>
                Третий вариант
              </span>
            </button>
            <input
              autocomplete="off"
              class="hiddenInput"
              type="checkbox"
            />
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component tag vertical"
    >
      <span
        class="label"
      >
        Заголовок группы
      </span>
      <div
        class="checkboxList"
      >
        <label
          class="checkbox tagLabel"
        >
          <button
            class="component s my-tag"
            type="button"
          >
            <span>
              Первый вариант
            </span>
          </button>
          <input
            autocomplete="off"
            class="hiddenInput"
            type="checkbox"
          />
        </label>
        <label
          class="checkbox tagLabel"
        >
          <button
            class="component s"
            type="button"
          >
            <span>
              Второй вариант
            </span>
          </button>
          <input
            autocomplete="off"
            class="hiddenInput"
            type="checkbox"
          />
        </label>
        <label
          class="checkbox tagLabel"
        >
          <button
            class="component s"
            type="button"
          >
            <span>
              Третий вариант
            </span>
          </button>
          <input
            autocomplete="off"
            class="hiddenInput"
            type="checkbox"
          />
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox Display tests should display with children like boolean or string or others react children type correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component checkbox vertical"
      >
        <span
          class="label"
        >
          Заголовок группы
        </span>
        <div
          class="checkboxList"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component checkbox vertical"
    >
      <span
        class="label"
      >
        Заголовок группы
      </span>
      <div
        class="checkboxList"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
